cogs/Voting.py

```diff
@@ -0,0 +1,644 @@
+        """
+        Adds a role to the list of roles the vote will be sent to
+        If no roles are added, the vote will go to all users in a guild (unless a target voice channel has been set)
+        :param role: role id (e.g. 135496683009081345) or a role ping (e.g. @Student)
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        vote.add_role(role.id)
+        await ctx.send(f"Vote will be sent to those with the {role.name} role")
+
+    @is_vote_caller()
+    @vote.command(name="removeRole")
+    async def removeRole(self, ctx, *, role: discord.Role):
+        """
+           Removes a role to the list of roles the vote will be sent to
+           :param role: role id (e.g. 135496683009081345) or a role ping (e.g. @Student)
+           """
+        vote = self.vote_manager.get_vote(ctx)
+        vote.remove_role(role.id)
+        await ctx.send(f"Vote will no longer be sent to those with the {role.name} role")
+
+    @is_vote_caller()
+    @vote.command(name="setChair")
+    async def setChair(self, ctx, *, chair: discord.Member = None):
+        """
+        Sets the chair of a vote
+        If no chair defaults to sending the message to the channel the vote is closed in
+        :param chair: user id (e.g. 135496683009081345) or ping (e.g. @ito#8813)
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        if chair:
+            vote.set_chair(chair.id)
+            await ctx.send(f"Set chair to {chair.name}")
+        else:
+            vote.set_chair(None)
+            await ctx.send(f"Results will be sent to the channel vote is closed in")
+
+    @is_vote_caller()
+    @vote.command(name="setChannel")
+    async def setChannel(self, ctx, *, channel: discord.VoiceChannel = None):
+        """
+        Sets the target voice channel of a vote (Users connected to this channel will receive the vote message)
+        If there isn't one set votes will go to all users in a guild (unless target roles have been added)
+        :param channel: channel id (e.g. 135496683009081345) or mention (e.g. #cool-channel)
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        if channel:
+            vote.set_vc(channel.id)
+            await ctx.send(f"Set target channel to {channel.name}")
+        else:
+            vote.set_vc()
+            await ctx.send("Removed channel restriction on vote")
+
+    @is_vote_caller()
+    @vote.command(name="addOption")
+    async def addOption(self, ctx, *, option_string):
+        """
+        Adds an option to the current vote
+        separate the title and description with a "+" e.g. option title+option description
+        :param option_string: a title and description for the option separated by a '+'
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        if len(vote.options) > 9:
+            await ctx.send("Vote has maximum number of options already (10)")
+            return
+        if len(option_string) > 600:
```

Comment:
10 of length 600 will excede the 2k cap

options:
- split the vote into multiple 2k messages when sending
- evaluate if the total vote goes over 2k in one of these checks
- decrease this value

Reply:
will cap total option length to 1500 chars and title to 200 which will leave 300 for the message sent with the vote (usually around 150)